storm-core/src/jvm/org/apache/storm/daemon/StormCommon.java

```diff
@@ -0,0 +1,604 @@
+    }
+
+    public static String getTopologyNameById(String topologyId) {
+        String topologyName = null;
+        try {
+            topologyName = topologyIdToName(topologyId);
+        } catch (InvalidTopologyException e) {
+            LOG.error("Invalid topologyId=" + topologyId);
+        }
+        return topologyName;
+    }
+
+    /**
+     * Convert topologyId to topologyName. TopologyId = topoloygName-counter-timeStamp
+     *
+     * @param topologyId
+     * @return
+     */
+    public static String topologyIdToName(String topologyId) throws InvalidTopologyException {
+        String ret = null;
+        int index = topologyId.lastIndexOf('-');
+        if (index != -1 && index > 2) {
+            index = topologyId.lastIndexOf('-', index - 1);
+            if (index != -1 && index > 0)
+                ret = topologyId.substring(0, index);
+            else
+                throw new InvalidTopologyException(topologyId + " is not a valid topologyId");
+        } else
+            throw new InvalidTopologyException(topologyId + " is not a valid topologyId");
+        return ret;
+    }
+
+    public static String getStormId(IStormClusterState stormClusterState, final String topologyName) {
+        List<String> activeTopologys = stormClusterState.activeStorms();
+        IPredicate pred = new IPredicate<String>() {
+            @Override
+            public boolean test(String obj) {
+                return obj != null ? getTopologyNameById(obj).equals(topologyName) : false;
+            }
+        };
+        return Utils.findOne(pred, activeTopologys);
+    }
+
+    public static Map<String, StormBase> topologyBases(IStormClusterState stormClusterState) {
+        return _instance.topologyBasesImpl(stormClusterState);
+    }
+
+    protected Map<String, StormBase> topologyBasesImpl(IStormClusterState stormClusterState) {
+        List<String> activeTopologys = stormClusterState.activeStorms();
+        Map<String, StormBase> stormBases = new HashMap<String, StormBase>();
+        if (activeTopologys != null) {
+            for (String topologyId : activeTopologys) {
+                StormBase base = stormClusterState.stormBase(topologyId, null);
+                if (base != null) {
+                    stormBases.put(topologyId, base);
+                }
+            }
+        }
+        return stormBases;
+    }
+
+    public static void validateDistributedMode(Map conf) {
+        if (ConfigUtils.isLocalMode(conf)) {
+            throw new IllegalArgumentException("Cannot start server in local mode!");
+        }
+    }
+
+    private static void validateIds(StormTopology topology) throws InvalidTopologyException {
+        List<String> componentIds = new ArrayList<String>();
+
+        for (StormTopology._Fields field : Thrift.getTopologyFields()) {
+            if (ThriftTopologyUtils.isWorkerHook(field) == false) {
+                Object value = topology.getFieldValue(field);
+                if (value != null) {
```

Comment:
Is this ever really null?  I don't see the null check even implicitly in the clojure code, if it is ever null I the clojure code would throw an NPE.